dev/NumberBox/NumberBox.cpp

```diff
@@ -0,0 +1,555 @@
+        {
+            newVal = MinValue();
+        }
+    }
+    
+    // Safeguard for floating point imprecision errors
+    int StepFreqSigDigits = ComputePrecisionRounderSigDigits(newVal);
+    m_stepPrecisionRounder.SignificantDigits(StepFreqSigDigits);
+    newVal = m_stepPrecisionRounder.RoundDouble(newVal);
+
+    // Update Text and Revalidate new value
+    Value(newVal);
+    UpdateTextToValue();
+    ValidateInput();
+}
+
+
+// Check if text resembles formulaic input to determine if parser should be executed
+bool NumberBox::IsFormulaic(const winrt::hstring& in)
+{
+    std::wstring input(in);
+    std::wregex formula(L"^([0-9()\\s]*[+-/*^%]+[0-9()\\s]*)+$");
+    std::wregex negval(L"^-([0-9])+(.([0-9])+)?$");
+    return (std::regex_match(input, formula) && !std::regex_match(input, negval));
+}
+
+// Computes the number of significant digits that precision rounder should use. This helps to prevent floating point imprecision errors. 
+int NumberBox::ComputePrecisionRounderSigDigits(double newVal)
+{
+    auto const oldVal = Value();
+    auto const stepFreq = StepFrequency();
+
+    // Run formatter on both values to discard trailing and leading 0's.
+    std::wstring formattedVal(m_stepPrecisionFormatter.Format(oldVal));
+    std::wstring formattedStep(m_stepPrecisionFormatter.Format(stepFreq));
+    std::wstring formattedNew(m_stepPrecisionFormatter.Format(newVal));
+
+    // Get size of only decimal portion of both old numbers. 
+    int oldValSig = static_cast<int>(formattedVal.substr(formattedVal.find_first_of('.') + 1).size());
+    int StepSig = static_cast<int>(formattedStep.substr(formattedStep.find_first_of('.') + 1).size());
+
+    // Pick bigger of two decimal sigDigits
+    int result = std::max(oldValSig, StepSig);
+    // append # of integer digits from new value
+    result += (int) formattedNew.substr(0, formattedNew.find_first_of('.')).size();
+    return result;
+}
+
+void NumberBox::NormalizeShorthandOperations()
+{
+    std::wregex r(L"^\\s*([+-/*^%]+[0-9()\\s]*)+$");
+    if (std::regex_match(m_TextBox.Text().data(), r))
+    {
+        std::wstringstream ss;
+        ss << Value() << m_TextBox.Text().data();
+        m_TextBox.Text(ss.str());
+    }
+}
+
+// Run value entered through NumberParser
+void NumberBox::EvaluateInput()
+{
+    std::optional<double> val;
+    val = NumberBoxParser::Compute(m_TextBox.Text());
+    // No calculation could be done
+    if (val == std::nullopt)
+    {
+        return;
+    }
+    if (std::fpclassify(val.value()) == FP_NAN)
+    {
+        Value(val.value());
+    }
+   Value(val.value());
+   UpdateTextToValue(); 
+}
+
+// Runs formatter and updates TextBox to it's value property, run on construction if Value != 0
+void NumberBox::UpdateTextToValue()
+{
+        winrt::hstring formattedValue(m_formatter.Format(Value()));
```

Comment:
[Prefer {} initializer syntax](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es23-prefer-the--initializer-syntax). And in cases like this it reads better as:
```suggestion
    auto formattedValue = m_formatter.Format(Value());
```